File: Laba1/Task8.cs
namespace PZPK_Labas.Laba1
{
    public static class Task8
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 8. Написати програму, згідно умови завдання з використанням вкладених інструкцій оператора for.");


            Console.WriteLine($"Сумма послідовності: {SumTask8()}");

            Console.ReadKey();
            Console.WriteLine();
        }

        private static double SumTask8()
        {
            double sum = 0;

            for (int x = 1; x <= 5; x++)
            {
                for (int k = 0; true; k++)
                {
                    double addAmount = Math.Pow(-1, k) * Math.Pow(x, k + 2) / ((k + 1) * Factorial(k + 2));

                    if (addAmount < 0.00000001)
                        break;

                    sum += addAmount;
                }
            }

            sum = Math.Round(sum, 3);

            return sum;
        }

        private static double Factorial(double number)
        {
            if (number == 0)
                return 1;
            else return number * Factorial(number - 1);
        }
    }
}

File: Laba1-Python/Task8.py
import math

def Run():
    print("Завдання 8. Написати програму, згідно умови завдання з використанням вкладених інструкцій оператора for.")

    print(f"Сума послідовності: {SumTask8()}")

    input()
    print()


def SumTask8():
    sum = 0

    for x in range(1, 6):
        for k in range(0, int(1e9)):
            add_amount = (pow(-1, k) * pow(x, k + 2)) / ((k + 1) * Factorial(k + 2))

            if add_amount < 1e-9:
                break

            sum += add_amount

    return round(sum, 3)


def Factorial(x):
    if x == 0:
        return 1
    else:
        return x * Factorial(x - 1)
